rct229/rules/section5/section5rule49.py

```diff
@@ -0,0 +1,132 @@
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):
+        rmr_proposed = context.proposed
+        return {"climate_zone": rmr_proposed["weather"]["climate_zone"]}
+
+    class BuildingRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section5Rule49.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, False, True),
+                required_fields={
+                    "$..zones[*]": ["infiltration", "surfaces"]
+                },
+            )
+
+        def get_calc_vals(self, context, data=None):
+            building_p = context.proposed
+
+            scc_dict_p = get_surface_conditioning_category_dict(
+                data["climate_zone"], building_p
+            )
+            zcc_dict_p = get_zone_conditioning_category_dict(
+                data["climate_zone"], building_p
+            )
+
+            building_total_envelope_area = 0.0
+            building_total_air_leakage_rate = 0.0
+            building_total_measured_air_leakage_rate = 0.0
+            empty_measured_air_leakage_rate_flow_flag = False
+
+            building_total_envelope_area = sum(
+                [
+                    surface.get("area")
+                    for surface in find_all("$..surfaces[*]", building_p)
+                    if scc_dict_p[surface["id"]] != SCC.UNREGULATED
+                ],
+                ZERO.AREA,
+            )
+
+            for zones in find_all("$..zones[*]", building_p):
+                if zones["infiltration"].get("measured_air_leakage_rate"):
+                    building_total_measured_air_leakage_rate += zones["infiltration"][
+                        "measured_air_leakage_rate"
+                    ]
+                else:
+                    empty_measured_air_leakage_rate_flow_flag = True
+                    if zcc_dict_p[zones["id"]] in [
+                        ZCC.CONDITIONED_RESIDENTIAL,
+                        ZCC.CONDITIONED_NON_RESIDENTIAL,
+                        ZCC.SEMI_HEATED,
+                    ]:
+                        building_total_air_leakage_rate += zones["infiltration"][
+                            "infiltration_flow_rate"
+                        ]
+
+            target_air_leakage_rate_75pa_p = (
+                TARGET_AIR_LEAKAGE_COEFF
+            ) * building_total_envelope_area
+
+            return {
+                "building_total_air_leakage_rate": building_total_air_leakage_rate,
+                "building_total_measured_air_leakage_rate": building_total_measured_air_leakage_rate,
+                "target_air_leakage_rate_75pa_p": target_air_leakage_rate_75pa_p,
+                "empty_measured_air_leakage_rate_flow_flag": empty_measured_air_leakage_rate_flow_flag,
+            }
+
```

Comment:
I think we need a `manual_check_required` method that checks for Case 2 in the Rule Assertion.